libnixf/src/Basic/diagnostic.py

```diff
@@ -184,10 +184,28 @@ class Diagnostic(TypedDict):
         "message": "undefined variable `{}`",
     },
     {
-        "sname": "sema-def-not-used",
-        "cname": "DefinitionNotUsed",
+        "sname": "sema-unused-def-let",
+        "cname": "UnusedDefLet",
+        "severity": "Warning",
+        "message": "variable `{}` in let-expression is not used",
```

Comment:
```suggestion
        "message": "definition `{}` in let-expression is not used",
```

This might be confusing with normal variables like `with pkgs; x`, both `pkgs` and `x` are called variables.